conda/resolve.py

```diff
@@ -902,17 +902,6 @@ def mysat(specs, add_if=False):
             solution, obj3 = C.minimize(eq_req_v, solution)
             log.debug('Initial package channel/version metric: %d/%d', obj3a, obj3)
```

Comment:
Instead of moving the feature handling, what if we add the following here:

```
# dependencies: maximize channel priority
eq_c, eq_v, eq_b, eq_t = r2.generate_version_metrics(C, speca)
solution, _ = C.minimize(eq_c, solution)
```

This way, we would (a) avoid that a package from a lower priority channel is chosen just to minimize updates and (b), avoid that we choose packages from a lower priority channel just to minimize features.

Reply:
Fine by me (at first thought). Will have to think about it a bit and try it out.
That might address test case T3  (https://github.com/conda/conda/issues/6199).
It might also resolve the issue in https://github.com/conda/conda/issues/6269 because there the offending feature-requiring package is in a lower priority channel. Although it wouldn't fix the underlying problem, i.e., test case T2, and thus https://github.com/conda/conda/issues/6269#issuecomment-345046410.

Reply:
True, but at least it won't span the problem across channels. E.g., that bug could be fixed in conda-forge by removing or masking the non-feature builds right?

And it is certainly more consistent with the docs and how channel priorities are described.

Reply:
Yes, I'd be a good thing to have this not affect higher priority channels. I'm not sure though how feasible it would be to remove/mask for all affected packages..

I'll try your changes in a bit, but I think those won't work either for certain use cases. For example take a user wanting `numpy` with the `defaults`' `nomkl` variety, but normally uses `conda-forge` packages, i.e., `conda-forge` has a higher priority. So they install the `nomkl` and `numpy` package and will most likely get `conda-forge`'s `numpy` of the `blas_openblas` variety instead of the `nomkl` one.
I really wish features would be something that gets highest priorities **iff** they're explicitly requested...

Reply:
I gave your suggestion a try:
* `conda create -n x -c conda-forge -c defaults numpy nomkl`
  Same behavior (and I don't know what the expected result should be):
  * `4.3.30`: installs `numpy:           1.13.3-py36_blas_openblas_201 conda-forge [blas_openblas]`
  * patched: installs `numpy:           1.13.3-py36_blas_openblas_201 conda-forge [blas_openblas]`
* `conda create -n x -c conda-forge pandas`:
  As expected it helps #6199:
  * `4.3.30`: installs `numpy:           1.13.3-py36ha12f23b_0`
  * patched: installs `numpy:           1.13.3-py36_blas_openblas_201 conda-forge [blas_openblas]`
* `conda create -n x -c bioconda -c conda-forge -c defaults trinity`
  Doesn't help #6269:
  * `4.3.30`: installs `trinity:             2.4.0-1            bioconda`
  * patched: installs `trinity:             2.4.0-1            bioconda`
* `conda create -n x -c bioconda -c conda-forge -c defaults rseqc`
  Helps with https://github.com/galaxyproject/tools-iuc/pull/1578:
  * `4.3.30`: installs `rseqc:           2.6.4-py27_0            bioconda`
  * patched: installs `rseqc:           2.6.4-py27r3.4.1_1            bioconda`

That results for `trinity`/`rseqc` were not what I initially suspected, but it makes sense why the behavior changed that way -- but it's still not what we overall want.